pkg/kubelet/dockertools/exec.go

```diff
@@ -96,10 +94,16 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		err = command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
 	}
-
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
```

Comment:
Have been wrong above about the timeout func being deferred, only the outer function call (`Stop()`) is (https://golang.org/ref/spec#Defer_statements). Still, being explicit here makes it much easier to read.